google/data_source_google_compute_vpn_gateway.go

```diff
@@ -0,0 +1,84 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func dataSourceGoogleComputeVpnGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleComputeVpnGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"network": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceGoogleComputeVpnGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	vpnGatewaysService := compute.NewTargetVpnGatewaysService(config.clientCompute)
+
+	gateway, err := vpnGatewaysService.Get(project, region, name).Do()
+	if err != nil {
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
```

Comment:
I think you want handleNotFoundError.

Reply:
This actually includes the 404 check - it returns the not-found error if the resource doesn't exist, and a regular error otherwise.  Just `if err != nil { return handleNotFoundError(...) }`

Reply:
I've just pushed another change. Does that work?
Also, I had modeled the original error handling code on existing examples such as `data_source_compute_network`, `data_source_compute_subnetwork` or `data_source_compute_global_address`. There might be other such examples. Are those incorrect as well?